webapp/packages/plugin-data-viewer-result-trace-details/src/useResultTraceDetails.tsx

```diff
@@ -42,44 +41,41 @@ export function useResultTraceDetails(model: IDatabaseDataModel<any, IDatabaseRe
       get trace(): DynamicTraceProperty[] | undefined {
         return this.cache.get(FAKE_ELEMENT_KEY, RESULT_TRACE_DETAILS_CACHE_KEY);
       },
-      loading: false,
+      promise: null,
       exception: null,
-      isLoading() {
-        return this.loading;
-      },
       isError() {
         return isContainsException(this.exception);
       },
       isLoaded() {
         return this.trace !== undefined;
       },
       async load() {
-        if (this.loading) {
-          return;
-        }
-
         const result = this.model.getResult(this.resultIndex);
 
         try {
           if (!result?.id) {
             throw new Error('Result is not found');
           }
 
-          this.loading = true;
+          if (this.promise) {
+            return;
+          }
+
           this.exception = null;
 
-          const { trace } = await dvResultTraceDetailsService.getTraceDetails(result.projectId, result.connectionId, result.contextId, result.id);
+          this.promise = dvResultTraceDetailsService.getTraceDetails(result.projectId, result.connectionId, result.contextId, result.id);
+          const { trace } = await this.promise;
```

Comment:
this promise is not stable and may lead to infinite component refresh. Stable promise is a promise that stored and can be accessed outside of the component it suspends.
It also may lead to multiple concurrent requests to the backend because state of component is lost when it's suspended